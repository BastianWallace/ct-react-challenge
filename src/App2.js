
function App2() {

  return (
    <div className="App">
      Hello World!
    </div>
  )
}

export default App2
