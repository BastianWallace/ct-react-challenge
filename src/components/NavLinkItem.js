import React from 'react'
import { useSelector, useDispatch } from 'react-redux'
import { changeContent } from '../store/slices/content'

const NavLinkItem = (props) => {
  const { item } = props
  const dispatch = useDispatch()
  const { content } = useSelector(state => state.content)

  const handleChangeContent = (event, content) => {
    event.preventDefault()
    dispatch(changeContent(content))
  }

  return (
    <li className="nav-item">
      {content === item.content && (
        <a type="button" className="nav-link active" aria-current="page">{item.title}</a>
      ) || (
        <a type="button" className="nav-link" onClick={event => handleChangeContent(event, item.content)} href="#">{item.title}</a>
      )}
    </li>
  )
}

export default NavLinkItem
