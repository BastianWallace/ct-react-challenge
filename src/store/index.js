import { configureStore } from '@reduxjs/toolkit'

// reducers
import content from './slices/content'
import categories from './slices/categories'
import favorites from './slices/favorites'

// export default configureStore({
//   reducer: {
//     content,
//     categories,
//     favorites
//   }
// })

export const setupStore = preloadedState => {
  return configureStore({
    reducer: {
      content,
      categories,
      favorites
    },
    preloadedState
  })
}