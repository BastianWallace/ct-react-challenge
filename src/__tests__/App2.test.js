/*
import { render } from '@testing-library/react'
import { Provider } from 'react-redux'
import { setupStore } from './store'
import App2 from './App2'
*/
import { Provider } from 'react-redux'
import { setupStore } from '../store'



import React from 'react'
//import { rest } from 'msw'
//import { setupServer } from 'msw/node'
import { fireEvent, screen, render, waitFor } from '@testing-library/react'
import renderer from 'react-test-renderer'
// We're using our own custom render function and not RTL's render.
//import { renderWithProviders } from './utils/test-utils'
//import App2 from '../App2'
import NavLinkItem from '../components/NavLinkItem'

// We use msw to intercept the network request during the test,
// and return the response 'John Smith' after 150ms
// when receiving a get request to the `/api/user` endpoint
// export const handlers = [
//   rest.get('/api/user', (req, res, ctx) => {
//     return res(ctx.json('John Smith'), ctx.delay(150))
//   })
// ]

//const server = setupServer(...handlers)

// Enable API mocking before tests.
//beforeAll(() => server.listen())

// Reset any runtime request handlers we may add during the tests.
//afterEach(() => server.resetHandlers())

// Disable API mocking after the tests are done.
//afterAll(() => server.close())

test('testing the entire App', async () => {
  //const a = renderWithProviders(<App2 />)
  const a = renderer.create(
    <Provider store={setupStore({})}>
      <NavLinkItem item={{title: 'Home', content: 'home'}} />
    </Provider>
  )

  await waitFor(() => {
    expect(a).toMatchSnapshot()
  })

  // should show no user initially, and not be fetching a user
  //expect(screen.getByText(/no user/i)).toBeInTheDocument()
  //expect(screen.queryByText(/Fetching user\.\.\./i)).not.toBeInTheDocument()

  // after clicking the 'Fetch user' button, it should now show that it is fetching the user
  //fireEvent.click(screen.getByRole('button', { name: /Fetch user/i }))
  //expect(screen.getByText(/no user/i)).toBeInTheDocument()

  // after some time, the user should be received
  //expect(await screen.findByText(/John Smith/i)).toBeInTheDocument()
  //expect(screen.queryByText(/no user/i)).not.toBeInTheDocument()
  //expect(screen.queryByText(/Fetching user\.\.\./i)).not.toBeInTheDocument()
})
