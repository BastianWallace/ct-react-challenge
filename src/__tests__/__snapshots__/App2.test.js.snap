// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the entire App 1`] = `
<div
  className="App"
>
  Hello World!
</div>
`;
